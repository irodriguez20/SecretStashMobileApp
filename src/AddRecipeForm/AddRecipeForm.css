.AddRecipe {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.AddRecipe__header {
  color: #443843;
}

.AddRecipe h1 {
  margin-left: 18px;
}

@media only screen and (max-width: 380px) {
  .AddRecipe h1 {
    text-align: center;
  }
}
