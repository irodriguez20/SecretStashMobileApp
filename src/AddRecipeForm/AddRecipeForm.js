import React, { Component } from 'react';
import Context from '../Context'
import SecretStashForm from '../SecretStashForm/SecretStashForm'
import './AddRecipeForm.css';
import PropTypes from 'prop-types';
import config from '../config';

class AddRecipeForm extends Component {
    static propTypes = {
        history: PropTypes.shape({
            push: () => { }
        }).isRequired,
    };

    static contextType = Context;

    handleSubmit = e => {
        e.preventDefault();

        const { name, folderid, timetomake, description, ingredients, steps } = e.target;
        const newRecipe = {
            id: 4,
            name: name.value,
            folderid: folderid.value,
            timetomake: timetomake.value,
            description: description.value,
            ingredients: ingredients.value,
            steps: steps.value,
        }

        fetch(`${config.API_ENDPOINT}/recipes`, {
            method: "POST",
            body: JSON.stringify(newRecipe),
            headers: {
                "content-type": "application/json"
            }
        })
            .then(res => {
                if (!res.ok) {
                    return res.json().then(e => Promise.reject(e));
                }
                return res.json();
            })
            .then(recipe => {
                name.value = "";
                timetomake.value = "";
                folderid.value = "";
                description.value = "";
                ingredients.value = "";
                steps.value = "";
                this.context.addRecipe(recipe)
                this.props.history.push('/');
            })
            .catch(error => {
                console.error({ error });
            });
    };

    render() {
        const { folders = [] } = this.context;
        return (
            <section className="AddRecipe">
                <header className="AddRecipe">
                    <h1>Add a New Memory</h1>
                </header>
                <SecretStashForm onSubmit={this.handleSubmit}>
                    <div className="field">
                        <label htmlFor="recipe-name-input">
                            Recipe Name
                            {' '}
                        </label>
                        <input
                            type="text"
                            name='name'
                            id='name'
                            required
                        />
                    </div>
                    <div className="field">
                        <label htmlFor="recipe-timetomake-input">
                            Estimated timetomake
                            {' '}
                        </label>
                        <input
                            type='text'
                            name='timetomake'
                            id='timetomake'
                            required
                        />
                    </div>
                    <div className="field">
                        <label htmlFor='recipe-folder-select'>
                            Folder
                            {' '}
                        </label>
                        <select id="folderid" name="folderid" required>
                            <option value={null}>...</option>
                            {folders.map(folder => (
                                <option key={folder.id} value={folder.id} name="folderid">
                                    {folder.foldername}
                                </option>
                            ))}
                        </select>
                    </div>
                    <div className="field">
                        <label htmlFor="recipe-description-textarea">
                            Description
                            {' '}
                        </label>
                        <textarea
                            name='description'
                            id='description'
                            required
                        ></textarea>
                    </div>
                    <div className="field">
                        <label htmlFor="recipe-ingredients-textarea">
                            Ingredients
                            {' '}
                        </label>
                        <textarea
                            rows='7'
                            name='ingredients'
                            id='ingredients'
                            required
                        ></textarea>
                    </div>
                    <div className="field">
                        <label htmlFor="steps">
                            Steps
                            {' '}
                        </label>
                        <textarea
                            rows='15'
                            name='steps'
                            id='steps'
                            required></textarea>
                    </div>
                    <div className='buttons'>
                        <button type='submit'>Add</button>
                    </div>
                </SecretStashForm>
            </section>
        )
    }
}

export default AddRecipeForm;