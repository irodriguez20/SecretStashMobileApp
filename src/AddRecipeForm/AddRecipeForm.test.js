import React from "react";
import AddRecipeForm from "./AddRecipeForm";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe(`AddRecipeForm component`, () => {
    const stubFolders = [
        {
            id: "1",
            name: "Breakfast"
        },
        {
            id: "2",
            name: "Lunch"
        },
        {
            id: "3",
            name: "Dinner"
        }
    ];

    it("renders the complete form", () => {
        const wrapper = shallow(<AddRecipeForm />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it.skip("renders the select options from folders", () => {
        const context = { folders: stubFolders };
        const select = shallow(<AddRecipeForm />, context).find("#recipe-folder-select");
        expect(toJson(select)).toMatchSnapshot();
    });
});
