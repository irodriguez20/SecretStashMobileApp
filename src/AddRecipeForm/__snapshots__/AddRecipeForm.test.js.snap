// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRecipeForm component renders the complete form 1`] = `
<section
  className="AddRecipe"
>
  <header
    className="AddRecipe__header"
  >
    <h1>
      Add a New Memory
    </h1>
  </header>
  <SecreStashForm
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <label
        htmlFor="recipe-name-input"
      >
        Recipe Name
         
      </label>
      <input
        id="name"
        name="name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="recipe-timetomake-input"
      >
        Estimated timetomake
         
      </label>
      <input
        id="timetomake"
        name="timetomake"
        required={true}
        type="text"
      />
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="recipe-folder-select"
      >
        Folder
         
      </label>
      <select
        id="folderid"
        name="folderid"
        required={true}
      >
        <option
          value={null}
        >
          ...
        </option>
      </select>
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="recipe-description-textarea"
      >
        Description
         
      </label>
      <textarea
        id="description"
        name="description"
        required={true}
      />
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="recipe-ingredients-textarea"
      >
        Ingredients
         
      </label>
      <textarea
        id="ingredients"
        name="ingredients"
        required={true}
        rows="7"
      />
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="steps"
      >
        Steps
         
      </label>
      <textarea
        id="steps"
        name="steps"
        required={true}
        rows="15"
      />
    </div>
    <div
      className="buttons"
    >
      <button
        type="submit"
      >
        Add
      </button>
    </div>
  </SecreStashForm>
</section>
`;
